hubspot/tasks.py

```diff
@@ -68,14 +77,57 @@ def check_hubspot_api_errors():
         defaults={"checked_on": now_in_utc()}
     )
     last_timestamp = hubspot_timestamp(last_check.checked_on)
+
+    order_sync_cache = {}
+
     for error in get_sync_errors():
-        if error.get("errorTimestamp") > last_timestamp:
-            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
-                obj_type=error.get("objectType", "N/A"),
-                obj_id=error.get("integratorObjectId", "N/A"),
-                details=error.get("details", ""),
+        error_timestamp = error.get("errorTimestamp")
+        if error_timestamp > last_timestamp:
+            obj_type = (error.get("objectType", "N/A"),)
+            obj_id = parse_hubspot_id(error.get("integratorObjectId", "")) or "N/A"
+            error_type = error.get("type", "N/A")
+            details = error.get("details", "")
+            # Handle line syncs that may have failed because of pending order syncs
+            if "LINE_ITEM" in obj_type and error_type == "INVALID_ASSOCIATION_PROPERTY":
+                order_id = parse_hubspot_id(ASSOCIATED_DEAL_RE.search(details).group(1))
+                try:
+                    line = Line.objects.get(id=obj_id)
+                except ObjectDoesNotExist:
+                    log.error(
+                        "Missing LINE %s for hubspot error %s: %s",
+                        obj_id,
+                        error_type,
+                        details,
+                    )
+                    continue
+
+                if str(order_id) not in order_sync_cache:
+                    try:
+                        # Get sync data and cache it to check against potential other line errors
+                        sync_data = get_sync_status("DEAL", line.order.id)
+                        order_sync_cache = sync_data["lastProcessedAt"]
+                    except HTTPError as sync_status_error:
+                        if sync_status_error.response.status_code == 400:
+                            # 400 status indicates order is pending, ignore the error for now
+                            continue
+                        else:
+                            log.error(sync_status_error)
+                            continue
+
+                if (
+                    order_id != line.order.id
+                    or order_sync_cache[str(order_id)] > error_timestamp
+                ):
+                    # If the line's order has changed or the order has synced since the error, resync the line
+                    sync_line_item_with_hubspot.si(line.id)
+
+            log.error(
```

Comment:
Don't log an error here if a `sync_line_item_with_hubspot ` request is made above (use `continue` to avoid it).